applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1235,18 +1294,22 @@ static struct bt_bap_unicast_client_cb unicast_client_cbs = {
 	.discover = discover_cb,
 };
 
-static void disconnected_cleanup(uint8_t chan_idx)
+static void disconnected_cleanup(struct stream_index stream_idx)
 {
-	unicast_servers[chan_idx].device_conn = NULL;
-	unicast_servers[chan_idx].sink_ep = NULL;
-	memset(unicast_servers[chan_idx].sink_codec_cap, 0,
-	       sizeof(unicast_servers[chan_idx].sink_codec_cap));
-	unicast_servers[chan_idx].source_ep = NULL;
-	memset(unicast_servers[chan_idx].source_codec_cap, 0,
-	       sizeof(unicast_servers[chan_idx].source_codec_cap));
-
-	unicast_servers[chan_idx].num_sink_eps = 0;
-	unicast_servers[chan_idx].num_source_eps = 0;
+	uint8_t lvl1_idx = stream_idx.lvl1_idx;
+	uint8_t lvl2_idx = stream_idx.lvl2_idx;
+	uint8_t lvl3_idx = stream_idx.lvl3_idx;
+
+	unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].device_conn = NULL;
+	unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_ep = NULL;
+	memset(unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_codec_cap, 0,
+	       sizeof(unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_codec_cap));
+	unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].source_ep = NULL;
+	memset(unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].source_codec_cap, 0,
+	       sizeof(unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].source_codec_cap));
+
+	unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].num_sink_eps = 0;
+	unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].num_source_eps = 0;
```

Comment:
```suggestion
       struct le_audio_unicast_server *unicast_server = &unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx];

        unicast_server->device_conn = NULL;
        unicast_server->sink_ep = NULL;
        memset(unicast_server->sink_codec_cap, 0,
	       sizeof(unicast_server->sink_codec_cap));
        unicast_server->source_ep = NULL;
        memset(unicast_server->source_codec_cap, 0,
	       sizeof(unicast_server->source_codec_cap));

        unicast_server->num_sink_eps = 0;
        unicast_server->num_source_eps = 0;
```
Repeating myself, this was just an easy spot to show the difference :)